x-pack/plugins/security/server/saved_objects/secure_saved_objects_client_wrapper.test.ts

```diff
@@ -392,6 +408,26 @@ describe('#addToNamespaces', () => {
     // this operation is unique because it requires two privilege checks before it executes
     await expectObjectNamespaceFiltering(client.addToNamespaces, { type, id, namespaces }, 2);
   });
+
+  test(`adds audit event when successful`, async () => {
```

Comment:
The existing tests here rely on shared assertions above which includes checking that the legacy audit logger was invoked correctly ([example](https://github.com/elastic/kibana/blob/1071596e66acb93ff2ad5a86fb19fb43fa0d732d/x-pack/plugins/security/server/saved_objects/secure_saved_objects_client_wrapper.test.ts#L119-L121)).

Would it be posible to leverage those existing assertions to verify that the new audit logger was also invoked correctly? That would eliminate the need for dedicated audit logging tests for each operation, while also making sure that we have a good deal of coverage for the various audit logging scenarios.

Reply:
I'll buy that argument. One benefit the "legacy" approach had was that it asserted we didn't log anything except for what we expected to log. In other words, we verified that we didn't log both a success and failure event for the same operation. I think it'd be worthwhile to update [`expectAuditEvent`](https://github.com/elastic/kibana/pull/74640/files/#diff-87d8d8eed61ba46ef7f5c312d76f0d319f4fcc35f62801a438ef7dc73ec92c75R181) to do the same if possible. I haven't evaluated the latest changes to see how feasible this is however

Reply:
> One benefit the "legacy" approach had was that it asserted we didn't log anything except for what we expected to log. In other words, we verified that we didn't log both a success and failure event for the same operation.

Good point. I think the simplest thing to do would be to assert the number of expected calls to the audit logger in each test case. E.g., add this to each test case:
```ts
expect(clientOpts.auditLogger.log).toHaveBeenCalledTimes(1);
```